scripts/zones/Temenos/npcs/Armoury_Crate.lua

```diff
@@ -815,241 +1179,279 @@ local loot =
-    local  MimicID=0;
-    local X = npc:getXPos();
-    local Y = npc:getYPos();
-    local Z = npc:getZPos();
-
-    for coffer = 1,#ARMOURY_CRATES_LIST_TEMENOS,2 do
-        if (ARMOURY_CRATES_LIST_TEMENOS[coffer] == CofferID-16928768) then
-            CofferType=ARMOURY_CRATES_LIST_TEMENOS[coffer+1][1];
-            InstanceRegion=ARMOURY_CRATES_LIST_TEMENOS[coffer+1][2];
-            addtime=ARMOURY_CRATES_LIST_TEMENOS[coffer+1][3];
-            DespawnOtherCoffer=ARMOURY_CRATES_LIST_TEMENOS[coffer+1][4];
-            MimicID=ARMOURY_CRATES_LIST_TEMENOS[coffer+1][5];
-            lootID=ARMOURY_CRATES_LIST_TEMENOS[coffer+1][6];
-        end
-    end
-    local coffer = CofferID-16928768;
-
-    if (CofferType == cTIME) then
-        tpz.battlefield.ExtendTimeLimit(battlefield, addtime)
-    elseif (CofferType == cITEM) then
-        if (InstanceRegion == Central_Temenos_4th_Floor and coffer~=79) then
-            local randmimic = math.random(1,24)
-            if ( randmimic < 19) then
-                local MimicList={16928986,16928987,16928988,16928989,16928990,16928991,16928992,16928993,16928994,16928995,16928996,16928997,16928998,16928999,16929000,16929001,16929002,16929003};
-                GetMobByID(MimicList[randmimic]):setSpawn(X,Y,Z);
-                SpawnMob(MimicList[randmimic]):setPos(X,Y,Z);
-                GetMobByID(MimicList[randmimic]):updateClaim(player);
-            else
-                battlefield:setLocalVar("loot", 1)
-                battlefield:spawnLoot(npc)
-                tpz.battlefield.HandleLootRolls(battlefield, loot[lootID], nil, npc)
-            end
-            -- despawn les coffer du meme groupe
-            for coffer = 1, #ARMOURY_CRATES_LIST_TEMENOS, 2 do
-                if (ARMOURY_CRATES_LIST_TEMENOS[coffer+1][5] == MimicID) then
-                    GetNPCByID(16928768+ARMOURY_CRATES_LIST_TEMENOS[coffer]):setStatus(tpz.status.DISAPPEAR);
+    local crateID = npc:getID()
+    local model = npc:getModelId()
+    local X = npc:getXPos()
+    local Y = npc:getYPos()
+    local Z = npc:getZPos()
+    local bfid = battlefield:getID()
+    local hold = false
+    if npc:getLocalVar("open") == 0 then
+        switch (bfid): caseof 
+        {
+            [1298] = function() -- Temenos West Crate Handling
+                if crateID ~= ID.npc.TEMENOS_W_CRATE[7] then
+                    for i = 1, 6 do
+                        for j = 0, 2 do
+                            if crateID == ID.npc.TEMENOS_W_CRATE[i]+j then
+                                if model == 960 then
+                                    tpz.battlefield.HealPlayers(battlefield)
+                                elseif model == 961 then
+                                    tpz.limbus.handleLootRolls(battlefield, loot[bfid][i], nil, npc)
+                                elseif model == 962 then
+                                    tpz.limbus.extendTimeLimit(battlefield, 15, tpz.zone.TEMENOS)
+                                end
```

Comment:
This is a weird one, and it may not matter, but all 3 of these zones have different crate mechanics. East has 4 chests and if you get a mimic then there's weird mechanics there, the others have 3. West doesn't depop the other chests, and North has 3 chests and you can only pick 1.